Lib/test/test_pkgutil.py

```diff
@@ -176,6 +176,42 @@ def test_walkpackages_zipfile(self):
                 continue
             del sys.modules[pkg]
 
+    def test_walk_packages_with_same_package_name_in_pythonpath(self):
+        pkg1 = 'test_walk_packages_with_same_package_name_in_pythonpath'
+        pkg1_dir = os.path.join(self.dirname, pkg1)
+        os.mkdir(pkg1_dir)
+        f = open(os.path.join(pkg1_dir, '__init__.py'), "wb")
+        f.close()
+        os.mkdir(os.path.join(pkg1_dir, 'same_name'))
+        f = open(os.path.join(pkg1_dir, 'same_name', '__init__.py'), "wb")
+        f.close()
+        f = open(os.path.join(pkg1_dir, 'same_name', 'mod.py'), "wb")
+        f.close()
+
+        # Add another package in the path, with the same name as the one inside pkg1
+        pkg2 = 'same_name'
+        pkg2_dir = os.path.join(self.dirname, pkg2)
+        os.mkdir(pkg2_dir)
+        f = open(os.path.join(pkg2_dir, '__init__.py'), "wb")
+        f.close()
+        os.mkdir(os.path.join(pkg2_dir, 'test_same_name'))
+        f = open(os.path.join(pkg2_dir, 'test_same_name', '__init__.py'), "wb")
+        f.close()
+        f = open(os.path.join(pkg2_dir, 'test_same_name', 'another_mod.py'), "wb")
+        f.close()
+
+        expected = [
+            'same_name',
+            'same_name.mod'
+        ]
+        actual = [e[1] for e in pkgutil.walk_packages([os.path.join(self.dirname, pkg1)])]
+        self.assertEqual(actual, expected)
+
+        for pkg in expected:
```

Comment:
1. The cleanup should use `self.addCleaup()` to avoid side-effects when the test fails or is interrupted.
2. The cleanup should use `test.support.forget` to unload imported packages and modules, including `same_name.mod`.

Reply:
Fixed, thanks for pointing that out